sos/collector/sosnode.py

```diff
@@ -412,25 +412,29 @@ def check_sos_version(self, ver):
         :returns:   True if installed version is at least ``ver``, else False
         :rtype:     ``bool``
         """
-        def _format_version(ver):
-            # format the version we're checking to a standard form of X.Y.Z-R
+        def _format_version_to_pep440(ver):
+            """ Convert the version into a PEP440 compliant version scheme."""
+            public_version_re = re.compile(
+                    r"^([0-9][0-9.]*(?:(?:a|b|rc|.post|.dev)[0-9]+)*)\+?"
+                    )
             try:
-                _fver = ver.split('-')[0]
-                _rel = ''
-                if '-' in ver:
-                    _rel = '-' + ver.split('-')[-1].split('.')[0]
-                if len(_fver.split('.')) == 2:
-                    _fver += '.0'
-
-                return _fver + _rel
+                _, public, local = public_version_re.split(ver, maxsplit=1)
+                if not local:
+                    return ver
+                sanitized_local = re.sub("[+~]+", ".", local).strip("-")
+                pep440_version = "%s+%s" % (public, sanitized_local)
```

Comment:
f-string here please.

Reply:
Whooops, gotta remind myself of this now :)

fixed on: [fceb5f3](https://github.com/sosreport/sos/commit/fceb5f393bbeb5c9da1fa63877a2f8d2294f223f)